vendor/engines/secure_rooms/app/services/secure_rooms/access_handlers/order_handler.rb

```diff
@@ -0,0 +1,30 @@
+module SecureRooms
+
+  module AccessHandlers
+
+    class OrderHandler
+
+      def self.process(occupancy)
+        return unless occupancy.orderable?
+
+        order = Order.create!(
+          account: occupancy.account,
+          user: occupancy.user,
+          facility: occupancy.facility,
+          created_by_user: occupancy.user,
+        )
+
+        order.order_details.create!(
+          product: occupancy.secure_room,
+          created_by_user: occupancy.user,
+          quantity: 1,
+        )
+
+        order
```

Comment:
We also need to attach the occupancy to the order detail. For reservations, we do this with a `has_one :reservation` on the order detail, and the reservation has an `order_detail_id` (nullable) column. I'm still unsure whether that was the right decision vs putting a nullable reservation_id on order details, but I think until it actually causes real problems we should stick with that pattern.